pkg/apis/v1alpha1/common_validations.go

```diff
@@ -156,3 +158,22 @@ func validateLegacySelector(selectors LegacySelector, validPaths map[string]bool
 
 	return nil
 }
+
+func (t *TemplateSpec) validate() error {
+	if t.Template == nil && t.Ytt == "" {
+		return fmt.Errorf("invalid template: must specify one of template or ytt, found neither")
+	}
+	if t.Template != nil && t.Ytt != "" {
+		return fmt.Errorf("invalid template: must specify one of template or ytt, found both")
+	}
+	if t.Template != nil {
+		obj := unstructured.Unstructured{}
+		if err := json.Unmarshal(t.Template.Raw, &obj); err != nil {
+			return fmt.Errorf("invalid template: failed to parse object metadata: %w", err)
```

Comment:
oh I think this message is from back when we used the PartialObject or something that actually parsed just the metadata - perhaps just "failed to parse object"?